eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -715,9 +763,272 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
           goto(NORMAL) using d.copy(channelUpdate = channelUpdate1) storing()
       }
 
+    case Event(cmd: CMD_SPLICE, d: DATA_NORMAL) =>
+      d.spliceStatus match {
+        case SpliceStatus.NoSplice =>
+          if (d.commitments.isIdle && d.commitments.params.channelFeatures.hasFeature(DualFunding)) {
+            val parentCommitment = d.commitments.latest
+            val targetFeerate = nodeParams.onChainFeeConf.feeEstimator.getFeeratePerKw(target = nodeParams.onChainFeeConf.feeTargets.fundingBlockTarget)
+            val fundingContribution = InteractiveTxBuilder.computeLocalContribution(
+              isInitiator = true,
+              commitment = parentCommitment.commitment,
+              spliceInAmount = cmd.additionalLocalFunding,
+              spliceOut = cmd.spliceOutputs,
+              targetFeerate = targetFeerate)
+            if (parentCommitment.localCommit.spec.toLocal + fundingContribution < parentCommitment.localChannelReserve) {
+              log.warning("cannot do splice: insufficient funds")
+              cmd.replyTo ! RES_FAILURE(cmd, InvalidSpliceRequest(d.channelId))
+              stay()
+            } else if (cmd.spliceOut_opt.map(_.scriptPubKey).exists(!MutualClose.isValidFinalScriptPubkey(_, allowAnySegwit = true))) {
+              log.warning("cannot do splice: invalid splice-out script")
+              cmd.replyTo ! RES_FAILURE(cmd, InvalidSpliceRequest(d.channelId))
+              stay()
+            } else {
+              log.info(s"initiating splice with local.in.amount=${cmd.additionalLocalFunding} local.in.push=${cmd.pushAmount} out.amount=${cmd.spliceOut_opt.map(_.amount).sum}")
+              val spliceInit = SpliceInit(d.channelId,
+                fundingContribution = fundingContribution,
+                lockTime = nodeParams.currentBlockHeight.toLong,
+                feerate = targetFeerate,
+                fundingPubKey = keyManager.fundingPublicKey(d.commitments.params.localParams.fundingKeyPath, parentCommitment.fundingTxIndex + 1).publicKey,
```

Comment:
The `fundingPubKey` provided here must match the one that we will use in the `InteractiveTxBuilder`. There is no degree of freedom, as the API could let us think. A bit awkward, but a mistake here will be caught in tests are sigs won't check out.